### PR TITLE
Move TentacleScriptDecoratorBuilder and friends into new common project

### PR DESCRIPTION
**Pull request by APErebus**
# Background

When adding new tests in #911, I realised we needed to share the `TentacleServiceDecoratorBuilder` from the `Tests.Integration` project. Because the `CommonTestUtils` is a NETFX 4.8 & NET 6.0 project, but Tentacle.Client and the integration tests are only Net 6.0, I had to create a new common project to put these in, `Octopus.Tentacle.Tests.Integration.Common`

# Results

This is basically just a cheese moving exercise of all the decorator code into this new test project.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/main/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [x] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [x] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/TentacleServiceDecoratorBuilderExtensions.cs [hunk 1]
```diff
@@ -1,39 +1,20 @@
...
     public static class TentacleServiceDecoratorBuilderExtensions
     {
-        public static TentacleServiceDecoratorBuilder RecordMethodUsages<TService>(this TentacleServiceDecoratorBuilder builder, out IRecordedMethodUsages recordedUsages)
-            where TService : class
-        {
-            var localMethodUsages = new MethodUsages();
-            recordedUsages = localMethodUsages;
-
-            return builder.RegisterProxyDecorator<TService>(service => MethodUsageProxyDecorator.Create(service, localMethodUsages));
-        }
```
Comment (APErebus) [inline source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/TentacleServiceDecoratorBuilderExtensions.cs:16]: These methods were moved into the shared library, but the other ones use the `TentacleConfigurationTestCase` which was not moved, hence why this file still remains

Comment (flin-8) [reply source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/TentacleServiceDecoratorBuilderExtensions.cs:16]: Is it worth moving `TentacleConfigurationTestCase` to common?

Comment (APErebus) [reply source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/TentacleServiceDecoratorBuilderExtensions.cs:16]: It's not applicable to the Kubernetes integration tests as it has information such as version, communication type, which runtime etc.

**Review by flin-8 — APPROVED** [review]
LGTM 🎉

**Review by rhysparry — APPROVED** [review]
Fine from SAST perspective, once the builds are all good.

**Comment by APErebus** [comment]
I spent a lot of time trying to fix the build issues and finally got there. 🎉

**Pull request merged**